src/Icicle/Source/ToCore/ToCore.hs

```diff
@@ -637,19 +638,21 @@ convertReduce xx
 
 
  | Case (Annot { annAnnot = ann, annResult = retty }) scrut patalts <- xx
- = do   scrut' <- convertReduce scrut
-        let (pats,alts) = unzip patalts
-        alts'  <- mapM convertReduce alts
+ = do   let (pats,alts) = unzip patalts
+
+        scrut' <- convertReduce scrut
+        pats'  <- convertCaseFreshenPats pats
+        alts'  <- mapM convertExp alts
+        alts'' <- hoist runFreshIdentity $ lift $ mapM (ANormal.anormal ()) alts'
```

Comment:
my mantra: do the smallest, simplest thing that's correct. if it's generating excessive lets we can a-normalise it in a separate pass afterwards (we probably already do)